### PR TITLE
CORE-6393: Fix function names for client generated swagger definition

### PR DESCRIPTION
**Pull request by BenYip123**
- Remove "API" from RestResource names. This changes the class name for the generated client from `VirtualNodeAPIApi` to `VirtualNodeApi`
- Update swaggerBaselines

### CODE REVIEW

File: libs/rest/rest-server-impl/src/main/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/ResourceToOpenApiSpecMapper.kt [hunk 1]
```diff
@@ -281,10 +282,20 @@ private fun Class<*>.isNull(): Boolean {
...
+    val newName = if (retrieveApiVersionsSet(
+            RestApiVersion.C5_0,
+            RestApiVersion.C5_2
+        ).contains(apiVersion)
+    ) {
+        name
+    } else {
+        name.removeSuffix(" API")
+    }
+
```
Comment (vkolomeyko) [inline libs/rest/rest-server-impl/src/main/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/ResourceToOpenApiSpecMapper.kt]: Can we extract this into a private method and use in both places?

**Comment by BenYip123** [comment]
build e2e

**Review by nikinagy — APPROVED** [review]
LGTM (network)

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Comment by anien** [comment]
Just a question, why do we change the swagger baseline also for `v1`, `v5_1` and `v5_2` if APIs for those version have not changed?

**Comment by BenYip123** [comment]
> Just a question, why do we change the swagger baseline also for `v1`, `v5_1` and `v5_2` if APIs for those version have not changed?

It affects all API versions. It's only changing the header name on the swaggerUI, so it doesn't change functionality

**Comment by anien** [comment]
> > Just a question, why do we change the swagger baseline also for `v1`, `v5_1` and `v5_2` if APIs for those version have not changed?
> 
> It affects all API versions. It's only changing the header name on the swaggerUI, so it doesn't change functionality

Yes, I am aware it doesn't change the functionality. I was just curious why we even change the files that are used only in 5.0 and are released.
e.g `components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/CertificatesRestResource.kt`
has 
```
    minVersion = RestApiVersion.C5_0,
    maxVersion = RestApiVersion.C5_0
```
 As nothing really would change. I get it for the consistency though.

**Review by anien — APPROVED** [review]

**Pull request merged**